Src/DependencyCollector/Shared/Implementation/ApplicationInsightsUrlFilter.cs

```diff
@@ -9,52 +9,33 @@ internal sealed class ApplicationInsightsUrlFilter
         internal const string TelemetryServiceEndpoint = "https://dc.services.visualstudio.com/v2/track";
         internal const string QuickPulseServiceEndpoint = "https://rt.services.visualstudio.com/QuickPulseService.svc";
 
+        internal readonly Uri TelemetryServiceEndpointUri = new Uri(TelemetryServiceEndpoint);
         private readonly TelemetryConfiguration telemetryConfiguration;
 
         private KeyValuePair<string, string> cachedEndpointLeftPart = new KeyValuePair<string, string>();
 
         public ApplicationInsightsUrlFilter(TelemetryConfiguration telemetryConfiguration)
         {
             this.telemetryConfiguration = telemetryConfiguration;
+
+            // cache EndpointLeftPart ahaed, before first dependency call
+            this.GetEndpointLeftPart();
```

Comment:
nitpick: it reads a little strange to call a getter and ignore the result, especially when the comment says "cache" but the return value is not "cached".